### ISSUE TITLE
ICE: lexer accepted invalid byte literal \xA\x4zomg step 3

### ISSUE THREAD

**Issue by carllerche** [rust-lang/rust#16014]
I'm assuming the bug was having the following in my code: `b"\xA\x4zomg"`

Output:

```
/Users/carllerche/Code/tilde/skylight/agent/buffoon/src/input_stream.rs:185:39: 185:40 error: illegal character in numeric character escape: \\
/Users/carllerche/Code/tilde/skylight/agent/buffoon/src/input_stream.rs:185         let mut i = input_stream(b"\xA\x4zomg");
                                                                                                                  ^
error: internal compiler error: unexpected failure
note: the compiler hit an unexpected failure path. this is a bug.
note: we would appreciate a bug report: http://doc.rust-lang.org/complement-bugreport.html
note: run with `RUST_BACKTRACE=1` for a backtrace
task 'rustc' failed at 'lexer accepted invalid byte literal \xA\x4zomg step 3', /Users/rustbuild/src/rust-buildbot/slave/nightly-mac/build/src/libsyntax/ast_util.rs:784

stack backtrace:
   1:        0x1119c09c5 - rt::backtrace::imp::write::h670da5925f6b0d30c3p
   2:        0x1119c3e05 - failure::on_fail::hd2ee4dc5c2f2df93vjq
   3:        0x111c5a379 - unwind::begin_unwind_inner::h5463f49db654b0f6RPd
   4:        0x10fc6470f - unwind::begin_unwind::h7569371465659382885
   5:        0x10fcc794c - parse::byte_lit::he306bf8e7eb7bc7744R
   6:        0x10fccaedd - parse::binary_lit::h1c522a5ff6f61badn8R
   7:        0x10fcc74a0 - parse::parser::Parser<'a>::lit_from_token::hbfda2e5773e28cf5NlJ
   8:        0x10fccb3d3 - parse::parser::Parser<'a>::parse_lit::h32e595f9d4d27e14QnJ
   9:        0x10fccf43f - parse::parser::Parser<'a>::parse_bottom_expr::h001721d9236102cfJDJ
  10:        0x10fcd5377 - parse::parser::Parser<'a>::parse_dot_or_call_expr::ha7c9a7d7980e5303TVJ
  11:        0x10fcd8920 - parse::parser::Parser<'a>::parse_prefix_expr::hd9197dc6805fe0f9CdK
  12:        0x10fcd93a7 - parse::parser::Parser<'a>::parse_binops::h958547a2c978d5a4ZjK
  13:        0x10fcd9ccd - parse::parser::Parser<'a>::parse_assign_expr::h2a04ca0a868dbe2bqnK
  14:        0x10fcd6a18 - parse::parser::Parser<'a>::parse_dot_or_call_expr_with::closure.$x22closure$x22$LP$46294$RP$
  15:        0x10fcd2e24 - parse::parser::Parser<'a>::parse_seq_to_before_end::h15849659572484904368
  16:        0x10fcd66db - parse::parser::Parser<'a>::parse_unspanned_seq::h11534047932491626462
  17:        0x10fcd5ce2 - parse::parser::Parser<'a>::parse_dot_or_call_expr_with::hec749fb5e7ca6844bWJ
  18:        0x10fcd5388 - parse::parser::Parser<'a>::parse_dot_or_call_expr::ha7c9a7d7980e5303TVJ
  19:        0x10fcd8920 - parse::parser::Parser<'a>::parse_prefix_expr::hd9197dc6805fe0f9CdK
  20:        0x10fcd93a7 - parse::parser::Parser<'a>::parse_binops::h958547a2c978d5a4ZjK
  21:        0x10fcd9ccd - parse::parser::Parser<'a>::parse_assign_expr::h2a04ca0a868dbe2bqnK
  22:        0x10fcdb372 - parse::parser::Parser<'a>::parse_local::h57a28e23b6c55f1f5WK
  23:        0x10fcdb808 - parse::parser::Parser<'a>::parse_stmt::h567cef4acf90683aPZK
  24:        0x10fce6d1c - parse::parser::Parser<'a>::parse_block_tail_::hadfb7935f0f832edZaL
  25:        0x10fcbf723 - parse::parser::Parser<'a>::parse_inner_attrs_and_block::h72cfc0525ca7977ek9K
  26:        0x10fceb744 - parse::parser::Parser<'a>::parse_item_fn::h1987e79d04f00d33nOL
  27:        0x10fce09f9 - parse::parser::Parser<'a>::parse_item_or_view_item::h7cc5dd0b12e715c6OLM
  28:        0x10fce8172 - parse::parser::Parser<'a>::parse_items_and_view_items::h28d8e4194db72213JcN
  29:        0x10fcee265 - parse::parser::Parser<'a>::parse_mod_items::h0dc286f02a5e4a3a08L
  30:        0x10fce3591 - parse::parser::Parser<'a>::parse_item_or_view_item::h7cc5dd0b12e715c6OLM
  31:        0x10fce8172 - parse::parser::Parser<'a>::parse_items_and_view_items::h28d8e4194db72213JcN
  32:        0x10fcee265 - parse::parser::Parser<'a>::parse_mod_items::h0dc286f02a5e4a3a08L
  33:        0x10fcf2d64 - parse::parser::Parser<'a>::eval_src_mod::h47df463e8d6dbbe1NhM
  34:        0x10fce19cf - parse::parser::Parser<'a>::parse_item_or_view_item::h7cc5dd0b12e715c6OLM
  35:        0x10fce8172 - parse::parser::Parser<'a>::parse_items_and_view_items::h28d8e4194db72213JcN
  36:        0x10fcee265 - parse::parser::Parser<'a>::parse_mod_items::h0dc286f02a5e4a3a08L
  37:        0x10fcfaa24 - parse::parser::Parser<'a>::parse_crate_mod::hce91880b104f2e68lhN
  38:        0x10fd00054 - parse::parse_crate_from_file::hb2ebda3d100190f86wR
  39:        0x10f01683e - driver::driver::phase_1_parse_input::closure.$x22closure$x22$LP$128697$RP$
  40:        0x10ee418c7 - driver::driver::phase_1_parse_input::hb63fc060f639fef5pWt
  41:        0x10f00bbbe - driver::driver::compile_input::h98e023739b82992eNRt
  42:        0x10f0b59d4 - driver::run_compiler::h6548e0d7f0125a42cWw
  43:        0x10f0b3fa6 - driver::main_args::closure.$x22closure$x22$LP$134215$RP$
  44:        0x10f0c675b - task::TaskBuilder<S>::try_future::closure.$x22closure$x22$LP$135359$RP$
  45:        0x10f0c6650 - task::TaskBuilder<S>::spawn_internal::closure.$x22closure$x22$LP$135336$RP$
  46:        0x10fb3d51c - task::spawn_opts::closure.$x22closure$x22$LP$8299$RP$
  47:        0x111cbe86c - rust_try
  48:        0x111c572fb - unwind::try::h92598303908f37b1WDd
  49:        0x111c57063 - task::Task::run::h6d0a3920195d64bciPc
  50:        0x10fb3d3b1 - task::spawn_opts::closure.$x22closure$x22$LP$8244$RP$
  51:        0x111c59126 - thread::thread_start::h626866d803224533mcd
  52:     0x7fff8f63c772 - _pthread_start

**Comment by steveklabnik** [comment]
Yes:

```
fn main() {
    let x = b"\xA\x4zomg";
}
```

produces this ICE.

**Comment by JustAPerson** [comment]
**tl;dr**: *this bug appears to be fixed*
[@steveklabnik's example](https://github.com/rust-lang/rust/issues/16014#issuecomment-73821257) *does not ICE* but will instead give a valid error:

```
<anon>:2:18: 2:19 error: illegal character in numeric character escape: \\
<anon>:2     let x = b"\xA\x4zomg";
                          ^
error: aborting due to previous error
```

**Comment by alexcrichton** [comment]
Thanks @JustAPerson!
